test/integration-test.py

```diff
@@ -368,6 +370,159 @@ def test_http_challenge_timeout():
+
+    # Add an A record for the redirect target that sends it to the real chall
+    # test srv for a valid HTTP-01 response.
+    redirHostname = "pebble-challtestsrv.example.com"
+    challSrv.add_a_record(redirHostname, ["10.77.77.77"])
+
+    # Start a simple python HTTP server on port 5002 in its own thread.
+    # NOTE(@cpu): The pebble-challtestsrv binds 10.77.77.77:5002 for HTTP-01
+    # challenges so we must use the 10.88.88.88 address for the throw away
+    # server for this test and add a mock DNS entry that directs the VA to it.
+    redirect = "http://{0}/.well-known/acme-challenge/{1}".format(
+            redirHostname, token)
+    httpd = HTTPServer(('10.88.88.88', 5002), BouncerHTTPRequestHandler(redirect, bounceFirst))
+    thread = threading.Thread(target = httpd.serve_forever)
+    thread.daemon = False
+    thread.start()
+
+    def cleanup():
+        # Remove the challtestsrv mocks
+        challSrv.remove_a_record(hostname)
+        challSrv.remove_a_record(redirHostname)
+        challSrv.remove_http01_response(token)
+        # Shut down the HTTP server gracefully and join on its thread.
+        httpd.shutdown()
+        httpd.server_close()
+        thread.join()
+
+    # Wait for the server to be ready before returning
+    wait_for_server("http://10.88.88.88/up")
+    return hostname, cleanup
+
+def test_http_multiva_threshold_pass():
+    # Only config-next has remote VAs configured and is appropriate for this
+    # integration test.
+    if not default_config_dir.startswith("test/config-next"):
+        return
+
+    client = chisel.make_client()
+
+    # These values should match the config in `config-next/va.json`
+    remoteVAs = 2
+    maxFailures = 1
+
+    # Configure a bounceFirst value that will pass the multiVA threshold test.
+    bounceFirst = (remoteVAs - maxFailures) + 1
+
+    # Create a testing hostname and the multiva server setup. This will block
+    # until the server is ready. The returned cleanup function should be used to
+    # stop the server. The first bounceFirst requests to the server will be sent
+    # to the real challtestsrv for a good answer, the rest will get a bad
+    # answer.
+    hostname, cleanup = multiva_setup(client, bounceFirst)
+
+    try:
+        # With the maximum number of allowed remote VA failures the overall
+        # challenge should still succeed.
+        auth_and_issue([hostname], client=client, chall_type="http-01")
+    finally:
+        cleanup()
+
+def test_http_multiva_threshold_fail():
+    # Only config-next has remote VAs configured and is appropriate for this
+    # integration test.
+    if not default_config_dir.startswith("test/config-next"):
+        return
+
+    client = chisel.make_client()
+
+    # Configure a bounceFirst value that will fail the multiVA threshold test by
+    # only redirecting the primary VA.
+    bounceFirst = 1
+
+    # Create a testing hostname and the multiva server setup. This will block
```

Comment:
This comment is duplicated with the paragraph below.

Also, I notice you have the same comment in both `test_http_multiva_threshold_pass` and `test_http_multiva_threshold_fail`. To me, the suggests that instead we should move this to the main comment for `multiva_setup`. That will de-duplicate it, but will also make each test case shorter so it's easier to see the commonalities and differences.

Reply:
Good idea, thanks. Fixed in 9a7b8f6